qiskit_ibm_runtime/utils/noise_model.py

```diff
@@ -0,0 +1,187 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Utility functions for the qiskit aer Noise Model class."""
+
+from warnings import catch_warnings, filterwarnings
+from typing import Any, Dict, List
+import numpy as np
+
+from qiskit.circuit import Instruction
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit import Reset
+from qiskit.circuit.library.generalized_gates import PauliGate, UnitaryGate
+
+from qiskit_aer.noise.errors.quantum_error import QuantumError
+from qiskit_aer.noise.errors.readout_error import ReadoutError
+from qiskit_aer.noise.noiseerror import NoiseError
+from qiskit_aer.noise.noise_model import NoiseModel
+
+
+def from_dict(noise_dict: Dict[str, Any]) -> NoiseModel:
```

Comment:
Should a deprecation warning be added as discussed on Slack?
cc'ing @jyu00

Reply:
We don't need to deprecate the function itself, since utility functions are not public and can be removed at any time.

We should, however, issue a deprecation warning if [noise_model](https://github.com/Qiskit/qiskit-ibm-runtime/blob/8c612d1343c2b8fd800fd0562974be783c58bdaa/qiskit_ibm_runtime/options/simulator_options.py#L56) options is used outside of local testing mode, so we can eventually remove this `from_dict` function. But that can also be a separate PR.

Reply:
I'll handle that in a separate PR